src/main/java/net/imglib2/util/Util.java

```diff
@@ -860,13 +868,13 @@ else if ( b == 1 )
 	 */
 	public static < T extends NativeType< T > > ImgFactory< T > getArrayOrCellImgFactory( final Dimensions targetSize, final int targetCellSize, final T type )
 	{
-		if ( Intervals.numElements( targetSize ) <= Integer.MAX_VALUE )
+		if ( Intervals.numElements( targetSize ) <= MAX_ARRAY_SIZE )
```

Comment:
When adjusting the `cellSize` below, the `MAX_ARRAY_SIZE` (or formerly `Integer.MAX_VALUE`) is divided by `type.getEntitiesPerPixel().getRatio()`. Wouldn't it make sense to also consider `type.getEntitiesPerPixel().getRatio()` in the checks for the number of required array elements for `ArrayImg` and cell size? I.e.
 1. 
```java
if ( Intervals.numElements( targetSize ) * type.getEntitiesPerPixel().getRatio() <= MAX_ARRAY_SIZE )
```
 2. 
```java
if ( Math.pow( targetCellSize, targetSize.numDimensions() ) * type.getEntitiesPerPixel().getRatio() <= MAX_ARRAY_SIZE )
```

This is not necessarily responsibility of this PR but could be fixed while at it.

Reply:
@hanslovsky Thank you for pointing that out. I changed the code accordingly.